packages/grammar/src/Happy/Grammar.lhs

```diff
@@ -44,15 +40,8 @@ The Grammar data type.
 >               first_term        :: Name,
 >               eof_term          :: Name,
 >               priorities        :: [(Name,Priority)],
->               token_type        :: String,
->               imported_identity :: Bool,
->               monad             :: (Bool,String,String,String,String),
->               expect            :: Maybe Int,
 >               attributes        :: [(String,String)],
```

Comment:
Shouldn’t `attribute` and `attributetypes` also move to `CommonOptions`? They’re not used by `happy-tabular`

Reply:
Yes they should be moved out, but I figured I would deal with all attribute things in the next PR. 

Also I vaguely recall only one backend supported attribute grammars?

Reply:
Yeah, I think GLR doesn’t.

Reply:
OK Yeah so the attributes stuff should be moved out to a different data type, and with that change we should get some better errors if one tries to do attribute stuff with the GLR backend.

Reply:
I don’t know if GLR and attributes are fundamentally incompatible or if it’s a limitation of the current backend. In the latter case, we should probably still pass the attribute stuff to the backend and throw “NotImplemented” or something of this sort.

Reply:
happy’s documentation is unhelpfully vague on this point:

> Currently, attribute grammars cannot be generated for GLR parsers (It's not exactly clear how these features should interact...)

Reply:
Also I think if we had a separate datatype, the lack of attribute grammars during bootstrapping could perhaps become a little cleaner.

Reply:
@int-index between boostrapping and the uncertainty around GLR, are you fine leaving it as-is for now?

Reply:
Yes, fine with me. I was concerned that `happy-directives` is too fine-grained, but `happy-codegen-common` is reasonable.